cpp/CMakeLists.txt

```diff
@@ -11,6 +11,7 @@ catkin_package(
    LIBRARIES ${PROJECT_NAME})
 
 set(CMAKE_LIBRARY_PATH /opt/local/lib ${CMAKE_LIBRARY_PATH})
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
```

Comment:
@lounick nit: Position-independent code in static libraries is considered as bad practice, at least for [Debian packages](https://www.debian.org/doc/debian-policy/#libraries), for some good reasons. If you need to link the library into another shared library, you should simply remove the `STATIC` flag below or explicitly set it to `SHARED` (`add_library()` defaults to `SHARED` if `BUILD_SHARED_LIBS` is `ON`, which is the case when using catkin). Should still work on Android afaik, or otherwise the build system that builds the Android library must set `BUILD_SHARED_LIBS` to `OFF` explicitly or not use catkin. But what type of library to build should normally not be decided by the package itself or at least be a configurable option.

Reply:
Coorect. Thanks for the great explanation! :D